### PR TITLE
feat: Gpkg format support

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                      |
| New feature     | ✔                                                                      |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                      |

### CODE REVIEW

File: src/layers/models/layersManager.ts [hunk 1]
```diff
@@ -46,14 +51,10 @@ export class LayersManager {
...
-      grids: this.mapproxyConfig.cache.grids,
+      grids: this.mapproxyConfig.cache.grids.split(','),
       request_format: this.mapproxyConfig.cache.requestFormat,
       upscale_tiles: this.mapproxyConfig.cache.upscaleTiles,
-      cache: {
-        type: this.mapproxyConfig.cache.type,
-        directory: layerRequest.tilesPath,
-        directory_layout: this.mapproxyConfig.cache.directoryLayout,
-      },
+      cache: this.getCacheSource(layerRequest.tilesPath),
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:57]: get to variable before putting here

File: src/layers/models/layersManager.ts [hunk 2]
```diff
@@ -150,14 +151,10 @@ export class LayersManager {
...
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:157]: get to variable before putting here

File: src/layers/models/layersManager.ts [hunk 3]
```diff
@@ -150,14 +151,10 @@ export class LayersManager {
 
     const newCache: IMapProxyCache = {
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:152]: refactor addLayer and Update layer to use one function for creating newCache and newLayer -> check if other code can be refactored

**Review by asafMasa — CHANGES_REQUESTED** [review]
Add readme

File: src/containerConfig.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ function registerExternalValues(): void {
   container.register(Services.MAPPROXY, { useValue: mapproxyConfig });
   container.register(Services.FILEPROVIDER, {
     useFactory: (): IFileProvider => {
-      return mapproxyConfig.fileProvider === Providers.S3 ? new S3Provider(container) : new FSProvider();
+      return mapproxyConfig.fileProvider === Providers.S3 ? new S3Provider(container) : new FSProvider(container);
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/containerConfig.ts]: fixed in 0375c15

File: src/layers/models/layersManager.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts]: fixed in 0b9e76c

File: src/layers/models/layersManager.ts (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts]: fixed in 0b9e76c

File: src/layers/models/layersManager.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts]: fixed in 0b9e76c

File: config/default.json [hunk 5]
```diff
@@ -17,11 +17,12 @@
     "defaultFilePath": "config/default.yaml",
     "yamlFilePath": "/path/to/config/file.yaml",
     "cache": {
-      "grids": ["epsg4326dir"],
+      "grids": "epsg4326ul",
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply config/default.json:20]: README - TBD

**Review by asafMasa — APPROVED** [review]

**Pull request merged**